content/blog/webrtc-demo.md

```diff
@@ -0,0 +1,192 @@
+
+But the interesting part for the multiple calls handler is in the `Wazo.Phone.ON_REINVITE` event. We set up the `callSession` variable on the corresponding session id (we’re looking for it into our sessions object).
+
+```js
+  const initializeWebRtc = () => {
+    Wazo.Phone.connect({
+      media: {
+        audio: true,
+        video: true,
+      },
+    });
+
+    Wazo.Phone.on(Wazo.Phone.ON_CALL_INCOMING, openIncomingCallModal);
+    Wazo.Phone.on(Wazo.Phone.ON_CALL_ACCEPTED, onCallAccepted);
+    Wazo.Phone.on(Wazo.Phone.ON_CALL_FAILED, onCallFailed);
+    Wazo.Phone.on(Wazo.Phone.ON_CALL_ENDED, onCallEnded);
+    Wazo.Phone.on(Wazo.Phone.ON_CALL_HELD, onSessionUpdate);
+    Wazo.Phone.on(Wazo.Phone.ON_CALL_RESUMED, onSessionUpdate);
+    Wazo.Phone.on(Wazo.Phone.ON_CALL_MUTED, onSessionUpdate);
+    Wazo.Phone.on(Wazo.Phone.ON_CALL_UNMUTED, onSessionUpdate);
+    Wazo.Phone.on(
+      Wazo.Phone.ON_REINVITE,
+      (session, request, updatedCalleeName) => {
+        const callSession = sessions[session.id];
+        if (callSession) {
+          currentSession.realDisplayName = updatedCalleeName;
+
+          onSessionUpdate(currentSession);
+        }
+      },
+    );
+  }
+```
+
+The `updateScenes` function is called every time something is changed in our 
+scene. A scene is a call session : the interlocutor name, the buttons, etc… 
+The commented part is not relevant, so let’s take a look at what is happening from line 9 to 21. We are looking into the `sessions` object and for each session we are fetching : 
+
+* The corresponding session id
+* The session’s associated number
+
+We set up `currentSession` on the `callSession` id if the session is active (we’re on call with this session’s number). 
+
+```js
+  const updateScenes = status => {
+    const noActiveSession = !Object.keys(sessions).length;
+
+    $('#scenes').html('');
+    $('#calls-handler').html('');
+
+    ///[...]
+    
+    Object.keys(sessions).forEach(sessionId => {
+      const callSession = sessions[sessionId];
+      const newScene = addScene(callSession, callSession.cameraEnabled);
+      const isActive = currentSession.is(callSession);
+      const label = getNumber(callSession);
+
+      if (!isActive) {
+        newScene.hide();
+      }
+
+      const $bouton = $('#calls-handler').append(`<button type="button" data-sessionid="${sessionId}" class="btn switch-btn ${isActive ? 'active' : ''}">${label}</button>`);
+      $bouton.click(switchCall);
+
+    })
+  }
+```
+
+We also add a button corresponding to each session inside our calls handler HTML container (we will take a look at it later). This button calls an event named `switchCall`, let’s study it next.
+
+```js
+  const switchCall = event => {
+    event.stopImmediatePropagation();
+
+    const sessionId = $(event.target).attr('data-sessionid');
+    const callSession = sessions[sessionId];
+    
+    if (currentSession.is(callSession)) {
```

Comment:
By not showing the button for the active call, this check would be avoided.
But it's an improvement of the demo.